pkg/skaffold/sync/sync.go

```diff
@@ -154,7 +154,13 @@ func syncMapForArtifact(ctx context.Context, a *latestV1.Artifact, cfg docker.Co
 	case a.DockerArtifact != nil:
 		return docker.SyncMap(ctx, a.Workspace, a.DockerArtifact.DockerfilePath, a.DockerArtifact.BuildArgs, cfg)
 
-	case a.CustomArtifact != nil && a.CustomArtifact.Dependencies != nil && a.CustomArtifact.Dependencies.Dockerfile != nil:
+	case a.CustomArtifact != nil:
+		if a.CustomArtifact.Dependencies == nil {
+			return nil, build.ErrCustomBuildNoDependencies{}
+		}
+		if a.CustomArtifact.Dependencies.Dockerfile == nil {
+			return nil, build.ErrCustomBuildNoDockerfile{}
+		}
```

Comment:
In this case shouldn't we just have `ErrCustomBuildNoDockerfile`? This could be confusing for a user who runs into `ErrCustomBuildNoDependencies`, goes and adds a dependency that's not a Dockerfile, then runs into `ErrCustomBuildNoDockerfile`

Reply:
yeah i guess it's probably not out of the question that they'll hit one error followed by another different error which isn't a great UX. i'll update